src/services/ghost/GhostServiceManager.ts

```diff
@@ -256,6 +256,38 @@ export class GhostServiceManager {
 		await this.updateGlobalContext()
 	}
 
+	private async hasAccess(document: vscode.TextDocument) {
+		return document.isUntitled || (await this.initializeIgnoreController()).validateAccess(document.fileName)
+	}
+
+	public async codeSuggestion() {
+		if (!this.enabled) {
+			return
+		}
+		const editor = vscode.window.activeTextEditor
+		if (!editor) {
+			return
+		}
+
+		this.taskId = crypto.randomUUID()
+		TelemetryService.instance.captureEvent(TelemetryEventName.INLINE_ASSIST_AUTO_TASK, {
+			taskId: this.taskId,
+		})
+
+		const document = editor.document
+		if (!(await this.hasAccess(document))) {
+			return
+		}
+
+		// Ensure model is loaded
+		if (!this.model.loaded) {
+			await this.load()
+		}
```

Comment:
This await cannot impact prior checks?

Reply:
no, the previous commands are vscode stuff, this is model/autocomplete stuff